rasa/nlu/selectors/response_selector.py

```diff
@@ -642,8 +633,41 @@ def load(
 
         return model
 
+    def _uses_sequence_features_for_text(self) -> bool:
+        """Whether we make use of sequence features for the TEXT attribute.
+
+        Note that, just like the DIETClassifier, DIET2BOW can make use of sentence
+        features even if the number of transformer layers is set to 0 because it creates
+        a BOW representation from sequence+sentence features.
+
+        In contrast to that, DIET2DIET classifier uses the last token from the
+        sequence that is obtained by concatenating sequence and sentence features.
+        Hence, it only makes use of sequence features if and only if the number of
+        transformer layers is > 0.
+        """
+        return (
+            self.component_config[NUM_TRANSFORMER_LAYERS] > 0
+            or not self.use_text_as_label
```

Comment:
Because that flag determines whether DIET2DIET or DIET2BOW is used (see docstr).

Reply:
Hmm, ideally then we should use `self.model_class` here but that would again be too redundant because you will have to pass `self.use_text_as_label` as argument to it. Ideally `model_class` should be a property which uses `self.use_text_as_label` but then that would be problematic inside the `load` function. Barrage of redundancies 🤦

Reply:
I can add a reminder to the docstr :)

Reply:
done